samples/modules/lvgl/acceleration_chart/src/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Benjamin Cabé <benjamin@zephyrproject.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(app);
+
+static lv_obj_t *chart1;
+static lv_chart_series_t *ser_x;
+static lv_chart_series_t *ser_y;
+static lv_chart_series_t *ser_z;
+lv_timer_t *fetch_and_display_timer;
+
+#define ACCEL0_NODE DT_ALIAS(accel0)
+const struct device *display_dev;
+const struct device *sensor;
+
+
+/* Timer handler: fetches sensor data and append it to the chart */
+static void fetch_and_display(lv_timer_t *timer)
+{
+	struct sensor_value accel[3];
+	int rc = sensor_sample_fetch(sensor);
+
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_ACCEL_XYZ, accel);
+	}
+	if (rc < 0) {
+		LOG_ERR("ERROR: Update failed: %d\n", rc);
+	} else {
+		lv_chart_set_next_value(chart1, ser_x, sensor_value_to_double(&accel[0]));
+		lv_chart_set_next_value(chart1, ser_y, sensor_value_to_double(&accel[1]));
+		lv_chart_set_next_value(chart1, ser_z, sensor_value_to_double(&accel[2]));
+	}
+}
+
+void accelerometer_chart(lv_obj_t *parent)
+{
+	chart1 = lv_chart_create(parent);
+	lv_obj_set_size(chart1, LV_HOR_RES, LV_VER_RES);
+	lv_chart_set_type(chart1, LV_CHART_TYPE_LINE);
+	lv_chart_set_div_line_count(chart1, 5, 8);
+	lv_chart_set_range(chart1, LV_CHART_AXIS_PRIMARY_Y, -20, 20); /* roughly -/+ 2G */
+	lv_chart_set_update_mode(chart1, LV_CHART_UPDATE_MODE_SHIFT);
+
+	ser_x = lv_chart_add_series(chart1, lv_palette_main(LV_PALETTE_RED),
+				    LV_CHART_AXIS_PRIMARY_Y);
+	ser_y = lv_chart_add_series(chart1, lv_palette_main(LV_PALETTE_BLUE),
+				    LV_CHART_AXIS_PRIMARY_Y);
+	ser_z = lv_chart_add_series(chart1, lv_palette_main(LV_PALETTE_GREEN),
+				    LV_CHART_AXIS_PRIMARY_Y);
+
+	lv_chart_set_point_count(chart1, 50);
+
+	/* Do not display point markers on the data */
+	lv_obj_set_style_size(chart1, 0, LV_PART_INDICATOR);
+}
+
+int main(void)
+{
+	display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device not ready, aborting test");
+		return 0;
```

Comment:
Had that mistake in the other demo one also, I think returning `-ENODEV` would be more fitting.

Reply:
ahah yes I took some of your code and mistakes too, I guess. should have remembered about @fabiobaltieri's late comments on your PR.